src/sfCommand.ts

```diff
@@ -138,10 +161,65 @@ export abstract class SfCommand<T> extends Command {
     };
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  protected async catch(err: Error & { exitCode?: number }): Promise<any> {
+    process.exitCode = process.exitCode ?? err.exitCode ?? 1;
+    this.log(this.formatError(err));
+    if (this.jsonEnabled()) {
+      CliUx.ux.styledJSON(this.toErrorJson(err));
+    }
```

Comment:
We don't want to log anything but JSON if the user passes in `--json`, right?
```suggestion
    if (this.jsonEnabled()) {
      CliUx.ux.styledJSON(this.toErrorJson(err));
    } else {
      this.log(this.formatError(err));
    }
```

Reply:
@RodEsp `this.log(...)` in base class also guards for isJsonEnable, else is not needed.